### ISSUE TITLE
Classification predictions outside [0, 1]

### ISSUE THREAD

**Issue by jtrosenthal** [automl/auto-sklearn#775]
I am training a binary classifier and when calling `clf.predict_proba(X_test)` I get the following error:

```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-20-feb67b2139ae> in <module>
      1 predictions = clf.predict(X_test)
----> 2 predictions_proba = clf.predict_proba(X_test)[:,1]
      3 
      4 f1 = sklearn.metrics.f1_score(y_test, predictions)
      5 auc = sklearn.metrics.roc_auc_score(y_test, predictions_proba)

/opt/anaconda3/envs/autosklearn/lib/python3.7/site-packages/autosklearn/estimators.py in predict_proba(self, X, batch_size, n_jobs)
    716         assert(
    717             (pred_proba >= 0).all() and (pred_proba <= 1).all()
--> 718         ), "found prediction probability value outside of [0, 1]!"
    719 
    720         return pred_proba

AssertionError: found prediction probability value outside of [0, 1]!
```

I think this is similar to #585 but maybe that patch didn't cover predict_proba.
I am using autosklearn version 0.6.0.
Thanks for taking a look at this!

**Comment by iXanthos** [comment]
same error here. Any updates?

**Comment by mfeurer** [comment]
Hey @jtrosenthal and @iXanthos could you please share some details on the datasets used? Also, is this error very reproducible on your side? Could you please check the output of https://automl.github.io/auto-sklearn/master/api.html#autosklearn.classification.AutoSklearnClassifier.get_models_with_weights and paste it here, too?

**Comment by jtrosenthal** [comment]
Hi @mfeurer, thanks for the reply. My error was not reproducible - I restarted the kernel and tried running again and did not encounter the same error.

**Comment by mfeurer** [comment]
Okay, then I'll close this for now as this doesn't appear to be a common issue and we cannot reproduce it. If you (or someone else) encounters this again, could you please obtain the info I mentioned above and post it here?

**Comment by mfeurer** [comment]
This was fixed by @franchuterivera in #852. It was caused by a numerical imprecission when averaging the models, and results in probabilities such as `1.0000000000000002`. It will be available in the next release.
